### PR TITLE
VP-7240: Add StoreNotificationSenderUnitTests

### PR DESCRIPTION
**Pull request by eMazeika**
Cover StoreNotificationSender scenarios https://virtocommerce.atlassian.net/browse/VP-6892
https://virtocommerce.atlassian.net/browse/VP-7240

### CODE REVIEW

File: tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    public class StoreNotificationSenderUnitTests
+    {
+        const string _tokenUrlEncoded = "tokenWithSlashes%2FandPlusSigns%2BandDoubleEquals%3D%3D";
+
+        private readonly Mock<INotificationSearchService> _mockNotificationSearchService;
+        private readonly Mock<INotificationSender> _mocknotificationSender;
+        private readonly Mock<IStoreService> _mockStoreService;
+        private readonly CustomUserManager _mockUserManager;
+
+        public StoreNotificationSenderUnitTests()
```
Comment (Vectorfield4) [inline tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs:32]: The constructor is redundant.

File: tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                .Setup(ss => ss.SearchNotificationsAsync(It.IsAny<NotificationSearchCriteria>()))
+                .Returns(() =>
+                {
+                    var notificationSearchResult = new NotificationSearchResult { TotalCount = 1, Results = new[] { new ConfirmationEmailNotification { } } };
+                    return Task.FromResult(notificationSearchResult);
+                });
+
+
+            _mocknotificationSender.Setup(ss => ss.SendNotificationAsync(It.IsAny<Notification>())).Returns(Task.FromResult(new NotificationSendResult { IsSuccess = true }));
+            _mockStoreService.Setup(ss => ss.GetByIdAsync(It.IsAny<string>(), It.IsAny<string>())).Returns(Task.FromResult(GetStore()));
```
Comment (Vectorfield4) [inline tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs:121]: Can we use everywhere the `.ReturnAsync` method?

File: tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+            var result = new StoreNotificationSender(_mockNotificationSearchService.Object, _mocknotificationSender.Object, _mockStoreService.Object, () => _mockUserManager);
+            return result;
+
+        }
+
+        internal static CustomUserManager GetTestUserManager(Mock<IUserStore<ApplicationUser>> storeMock = null)
```
Comment (Vectorfield4) [inline tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs:128]: Please, add a comment above this line:
``` csharp
// TODO: PT-274 Move mock of AspNetUserManager to the platform
```

File: tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+            validator.Setup(x => x.ValidateAsync(userManager, It.IsAny<ApplicationUser>()))
+                .ReturnsAsync(IdentityResult.Success).Verifiable();
+
+            userManager.RegisterTokenProvider("Static", new StaticTokenProvider());
+            userManager.Options.Tokens.EmailConfirmationTokenProvider = "Static";
+            return userManager;
+        }
+    }
+
+    internal class CustomUserManager : AspNetUserManager<ApplicationUser>
```
Comment (Vectorfield4) [inline tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs:172]: Please, add a comment above this line:
``` csharp
// TODO: PT-274 Move mock of AspNetUserManager to the platform
```

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]
Please, add a link to the current Jira task (VP-7240).

File: tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs (hunk 1 above)
Comment (eMazeika) [reply tests/VirtoCommerce.StoreModule.Tests/StoreNotificationSenderUnitTests.cs:32]: ok

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**